staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go

```diff
@@ -336,27 +339,22 @@ func addEmbeddedProperties(s *spec.Schema) {
 	if s.Items != nil {
 		addEmbeddedProperties(s.Items.Schema)
 	}
+	if s.AdditionalProperties != nil {
+		addEmbeddedProperties(s.AdditionalProperties.Schema)
+	}
 	if isTrue, ok := s.VendorExtensible.Extensions.GetBool("x-kubernetes-embedded-resource"); ok && isTrue {
+		s.SetProperty("metadata", *spec.RefSchema(objectMetaSchemaRef).
```

Comment:
please verify what kubectl (kube-openapi) can cope with. I think a ref plus properties or so panics.

Reply:
the root metadata worked for [explain](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/crd_publish_openapi.go#L106-L109) and [(passes) client-side validation](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apimachinery/crd_publish_openapi.go#L65)

I will add a failure case and also for the embedded resources

Reply:
I see. I will check the behavior with a customized metadata (properties + $ref)